rs/nns/governance/src/governance.rs

```diff
@@ -4046,7 +4053,19 @@ impl Governance {
         //
         // See "Recommendations for Using `unsafe` in the Governance canister" in canister.rs
         let governance: &'static mut Governance = unsafe { std::mem::transmute(self) };
-        spawn(governance.perform_action(pid, action.clone()));
+
+        #[cfg(target_arch = "wasm32")]
+        {
+            spawn(governance.perform_action(pid, action.clone()));
+        }
+        // This is needed for tests
+        #[cfg(not(target_arch = "wasm32"))]
+        {
+            governance
+                .perform_action(pid, action.clone())
+                .now_or_never()
+                .expect("Future could not execute in non-WASM environment");
+        }
```

Comment:
I think what you can maybe do at the top of the file is this:

```
#[cfg(target_arch = "wasm32")]
use ic_cdk::spawn;
#[cfg(not(target_arch = "wasm32")]
use tokio::task::spawn;
```

(Ofc, you already have the first two lines.)

Then, there is no need to have conditional compilation HERE.

Reply:
Tokio is not a dependency in the wasm.  So this won't work.  The reason I picked what I picked is we already are using it in tests, and it doesn't need to work internally in the WASM.  Everything has to execute immediately because we don't have any actual source of asynchrony.